src/effects/rising_bubble.c

```diff
@@ -8,7 +8,43 @@ INCLUDE_ASM(s32, "effects/rising_bubble", rising_bubble_main);
 void rising_bubble_init(void) {
 }
 
-INCLUDE_ASM(s32, "effects/rising_bubble", rising_bubble_update);
+void rising_bubble_update(EffectInstance* effect) {
+    RisingBubbleFXData* risingBubbleFXData = effect->data.risingBubble;
+    s32 temp_a2 = risingBubbleFXData->unk_18;
+    s32 temp_v0;
+    s32 temp_v1;
+    s32 temp_v1_2;
+
+    if (risingBubbleFXData->unk_08 >= risingBubbleFXData->unk_24) {
+        risingBubbleFXData->unk_18--; 
+        risingBubbleFXData->unk_1C++;
+        
+        if (risingBubbleFXData->unk_18 < 0) {
+            shim_remove_effect(effect); 
+            return;
+        }
+        
+        if (temp_a2 >= 6) { 
+            risingBubbleFXData->unk_14 += (0x78 - risingBubbleFXData->unk_14) * 0.3;
+        } else {
+            risingBubbleFXData->unk_14 = (temp_a2 * 0x14);
+        }
+
+        risingBubbleFXData->unk_20++;
+        if (risingBubbleFXData->unk_20 >= 0x20) {
+            risingBubbleFXData->unk_20 -= 0x20;
+        }
+        
+        risingBubbleFXData->unk_10 += 0.1;
+        return;
+    }
+    
+    risingBubbleFXData->unk_14 = 0xFF;
```

Comment:
Please write all the hex numbers here in decimal. We only use hex for enums, pretty much, and decimal values are used for positions, measurements, color, etc